subsys/bluetooth/host/hci_core.c

```diff
@@ -2008,8 +2008,11 @@ static void hci_cmd_done(uint16_t opcode, uint8_t status, struct net_buf *buf)
 	if (cmd(buf)->opcode != opcode) {
 		BT_WARN("OpCode 0x%04x completed instead of expected 0x%04x",
 			opcode, cmd(buf)->opcode);
+		return;
 	}
 
+	bt_dev.sent_cmd = NULL;
+
```

Comment:
Can you elaborate on why this statement was moved? It isn't clear to me.

Reply:
see above.